python/GangaAtlas/Lib/ATLASDataset/DQ2Dataset.py

```diff
@@ -2124,6 +2124,9 @@ def addTask(self, cmd):
 # New for DQ2 client 2.3.0
 from Ganga.GPIDev.Credentials_old import GridProxy
 gridProxy = GridProxy()
+if not gridProxy.isValid():
+    gridProxy.create()
+
```

Comment:
For the new credential system, yes. However, I haven't had the time to properly sort that out yet so at present the code is still using the old `gridProxy` object to make sure a proxy is present at startup. In a future PR, this will be removed in favour of decorators on particular functions.